### ISSUE TITLE
Spark kill Java virtual machine on startup with Java OpenJ9

### ISSUE THREAD

**Issue by Chevels** [lucko/spark#372]
### Description

My server restarts in a loop as soon as I install Spark. Each time it generates a dump file of several GB and a javacore.20231020.074712.1.0002.txt file

I had to remove it a long time ago because we are running Java OpenJ9 for performance reasons (to fix docker issue with GC and RAM) and unfortunately Spark killed the Java machine on startup.
The Paper devs will abandon support for Timings to make way for Spark but if it is not OpenJ9 compatible some servers will no longer have effective profiling solutions :/

My server runs on Paper and has 86 plugins, everything has worked for years with Java OpenJ9, except Spark. As soon as I reinstall spark, the server no longer starts as explained above.

### Reproduction Steps

- shutdown server
- install Spark (.jar) in plugins folder
- start the server

### Expected Behaviour

That the server starts correctly with spark

### Platform Information

* Minecraft Version: 1.20.1
* Platform Type: server
* Platform Brand: Paper
* Platform Version: git-Paper-194 (MC: 1.20.1)

### Spark Version

v1.10.34

### Logs and Configs

[07:45:58] [Server thread/INFO]: [spark] Loading server plugin spark v1.10.34
[07:45:58] [Server thread/INFO]: [spark] Enabling spark v1.10.34
[07:45:58] [Server thread/INFO]: [spark] Using Paper ServerTickStartEvent for tick monitoring
[07:45:58] [Server thread/INFO]: [spark] Starting background profiler...
[07:45:58] [Server thread/INFO]: [PlaceholderAPI] Successfully registered expansion: spark
[07:45:58] [Server thread/INFO]: [spark] Registered PlaceholderAPI placeholders

As soon as Spark is enabled the log file no longer receives console logs and the server crashes instantly.
This is what the console displays at this time (but does not print to the .log file) :
https://gist.github.com/Chevels/9f9b9b687ef6ae6d609038f98c29219c

javacore.20231020.074712.1.0002.txt file generated on JVM crash:
https://gist.github.com/Chevels/0c985ad31d93712e0daf8a1cc4e889e7

### Extra Details

The problem occurs whether you start the server with Spark already installed or if you try to hot start it with plugman.

The problem does not depend on the version of Paper or the game because it has existed for a long time: I had already tried to install Spark in September 2022 (Paper for 1.19.2 Minecraft version) and the year before as well. Every time this error occurs if we use Java OpenJ9

Actually I'm using Java 17 OpenJ9 but it already did in Java 16 Openj9.

**Comment by lucko** [comment]
Resolved in 8379f2b07bc38c24733c4f1d537ea7c2ee03d91b

**Comment by Chevels** [comment]
> Resolved in [8379f2b](https://github.com/lucko/spark/commit/8379f2b07bc38c24733c4f1d537ea7c2ee03d91b)

Hello!
The problem has reappeared since I upgraded my server to 1.21.8. I'm not sure which version of Paper reintroduced the bug, as we upgraded directly from 1.21.4 to 1.21.8.

I'm using Paper's built-in Spark. I don't know if this is the right place to report it, but I think so.

javacore.20250803.130457.37.0002.txt.txt file generated on JVM crash:
[javacore.20250803.130457.37.0002.txt](https://github.com/user-attachments/files/21568721/javacore.20250803.130457.37.0002.txt)

**Server info:**
- Java 21 OpenJ9
- This server is running Paper version 1.21.8-21-main@ed31825 (2025-07-31T02:36:28Z) (Implementing API version 1.21.8-R0.1-SNAPSHOT)
- I don't know how to get the version of the Spark built into Paper. I haven't found a command to display the version.
